### PR TITLE
Fixing tar file saving, comparator for Graphs.jl, and the FileDFG tests

### PR DESCRIPTION
**Pull request by GearsAD**
REF: #305

### CODE REVIEW

File: src/FileDFG/services/FileDFG.jl [hunk 1]
```diff
@@ -11,18 +11,25 @@ dfg = initfg()
 # ... Add stuff to graph using either IIF or DFG:
 v1 = addVariable!(dfg, :a, ContinuousScalar, labels = [:POSE], solvable=0)
 # Now save it:
-saveDFG(dfg, "/tmp/saveDFG")
+saveDFG(dfg, "/tmp/saveDFG.tar.gz")
```
Comment (GearsAD) [inline src/FileDFG/services/FileDFG.jl]: This is to just subtly push users to use the extension so we can smoothly remove the folder code in the future.

Comment (dehann) [reply src/FileDFG/services/FileDFG.jl]: agree

**Comment by GearsAD** [comment]
FYI @pvazteixeira

**Comment by pvazteixeira** [comment]
thanks @GearsAD

**Comment by Affie** [comment]
See #308

**Comment by dehann** [comment]
Right, just avoid confusion -- we came to a decision how best to do compare functions (ie `==`and `isapprox`).  See #308

**Comment by dehann** [comment]
Hi @GearsAD , thanks for the fix!

**Pull request merged**